src/RA_httpthread.cpp

```diff
@@ -485,16 +484,18 @@ BOOL RAWeb::DoBlockingHttpPost(const std::string& sRequestedPage, const std::str
                     {
                         {
                             DWORD nBytesFetched = 0;
-                            if (auto pData{ std::make_unique<char[]>(nBytesToRead) }
-                            ; WinHttpReadData(hRequest, pData.get(), nBytesToRead, &nBytesFetched))
                             {
-                                ASSERT(nBytesToRead == nBytesFetched);
-                                ResponseOut.insert(ResponseOut.end(), pData.get(), pData.get() + nBytesFetched);
-                            }
-                            else
-                            {
-                                RA_LOG("Assumed timed out connection?!");
-                                break;  //Timed out?
+                                auto pData{ std::make_unique<char[]>(nBytesToRead) };
+                                if (WinHttpReadData(hRequest, pData.get(), nBytesToRead, &nBytesFetched))
+                                {
+                                    ASSERT(nBytesToRead == nBytesFetched);
+                                    ResponseOut.insert(ResponseOut.end(), pData.get(), pData.get() + nBytesFetched);
+                                }
+                                else
+                                {
+                                    RA_LOG("Assumed timed out connection?!");
+                                    break;  //Timed out?
+                                }
```

Comment:
the outer scope only contains the `nBytesFetched` variable, suggest eliminating inner scope.

Reply:
Pretty much put all your suggestions, for this one though I got rid of the outer scope (around the `DWORD`) and put that `DWORD `where `pData `is because the smart pointer definitely needs to get disposed of as soon as possible. It's not disposed of as early as it was before (either in the `if `or `else`, but figured uniformity of style was more preferrable.